mdb-realm/src/helpers.ts

```diff
@@ -1,126 +1,36 @@
-            if (process.platform === "win32") {
-                actualCmd = `pwsh -Command "${actualCmd.replace(/"/g, '\\"').split("\n").join("`n")}"`;
-            }
-            const response = await execCmd(actualCmd);
-            return response
-                .split(/\r?\n/)
-                .filter(s => s && s.trim() && !s.includes("Deploying app changes..."))
-                .map(s => JSON.parse(s));
-        } catch (error: any) {
-            if (retries-- < 2) {
-                throw error;
-            } else {
-                core.info(`Failed to execute ${cmd} with ${error}. Retrying ${retries} more time(s)`);
-            }
-        }
-    }
-}
-
-async function execAtlasRequest(
-    atlasUrl: string,
-    method: urllib.HttpMethod,
-    route: string,
-    config: EnvironmentConfig,
-    payload?: any,
-): Promise<any> {
-    const url = `${atlasUrl}/api/atlas/v1.0/groups/${config.projectId}/${route}`;
-
-    const request: urllib.RequestOptions = {
-        digestAuth: `${config.apiKey}:${config.privateApiKey}`,
-        method,
-        headers: {
-            "content-type": "application/json",
-            accept: "application/json",
-        },
-    };
-
-    if (payload) {
-        request.data = JSON.stringify(payload);
-    }
-
-    const response = await urllib.request(url, request);
-
-    if (response.status < 200 || response.status > 300) {
-        throw new Error(`Failed to execute ${request.method} ${route}: ${response.status}: ${response.data}`);
-    }
-
-    return JSON.parse(response.data);
-}
-
-function getSuffix(requireDifferentiator = true): string {
-    const differentiator = core.getInput("differentiator", { required: requireDifferentiator });
-    if (differentiator) {
-        return createHash("md5")
-            .update(`${getRunId()}-${differentiator}`)
-            .digest("base64")
-            .replace(/\+/g, "")
-            .replace(/\//g, "")
-            .toLowerCase()
-            .substring(0, 8);
-    }
-
-    return "no-cluster";
-}
-
-function getRunId(): string {
-    return process.env.GITHUB_RUN_ID || "";
+function generateClusterName(): string {
+    return createHash("md5")
+        .update(`${process.env.GITHUB_RUN_ID}-${process.env.GITHUB_RUN_ATTEMPT}`)
+        .digest("base64")
+        .replace(/\+/g, "")
+        .replace(/\//g, "")
+        .toLowerCase()
+        .substring(0, 8);
 }
 
-export function getConfig(requireDifferentiator = true): EnvironmentConfig {
+export function getConfig(): EnvironmentConfig {
     return {
         projectId: core.getInput("projectId", { required: true }),
         apiKey: core.getInput("apiKey", { required: true }),
         privateApiKey: core.getInput("privateApiKey", { required: true }),
         realmUrl: core.getInput("realmUrl", { required: false }) || "https://realm-dev.mongodb.com",
         atlasUrl: core.getInput("atlasUrl", { required: false }) || "https://cloud-dev.mongodb.com",
```

Comment:
The values of the default urls should point to the QA environment
```suggestion
realmUrl: core.getInput("realmUrl", { required: false }) || "https://realm-qa.mongodb.com",
atlasUrl: core.getInput("atlasUrl", { required: false }) || "https://cloud-qa.mongodb.com",
```

Reply:
Yeah, I forgot about this - will fix in a follow-up PR.